tf_agents/agents/dqn/dqn_agent.py

```diff
@@ -294,8 +294,7 @@ def _train(self, experience, weights):
       eager_utils.add_gradients_summaries(grads_and_vars,
                                           self.train_step_counter)
 
-    self._optimizer.apply_gradients(grads_and_vars,
-                                    global_step=self.train_step_counter)
+    self._optimizer.apply_gradients(grads_and_vars)
 
```

Comment:
add
# Increment train_step_counter
self.train_step_counter.assign_add(1)

Reply:
You are perfectly right @sguada. Thank you. Done.